app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt

```diff
@@ -275,78 +274,48 @@ class PresetsFragment : BaseFragment() {
     }
 
     inner class CategoriesPagerAdapter(fm: FragmentManager) :
-        FragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
+        VocableFragmentStateAdapter<Category>(fm, viewLifecycleOwner.lifecycle) {
 
-        private val categories = mutableListOf<Category>()
-
-        var numPages = 0
-
-        fun setCategories(categories: List<Category>) {
-            with(this.categories) {
-                clear()
-                addAll(categories)
-            }
-            numPages = ceil(categories.size / maxCategories.toDouble()).toInt()
-            notifyDataSetChanged()
-        }
-
-        override fun getItemCount(): Int {
-            return Int.MAX_VALUE
-        }
+        override fun getMaxItemsPerPage(): Int = maxCategories
 
         override fun createFragment(position: Int): Fragment {
             val startPosition = (position % numPages) * maxCategories
-            val subList = categories.subList(
+            val subList = items.subList(
                 startPosition,
-                min(categories.size, startPosition + maxCategories)
+                min(items.size, startPosition + maxCategories)
             )
 
             return CategoriesFragment.newInstance(subList)
         }
     }
 
     inner class PhrasesPagerAdapter(fm: FragmentManager) :
-        FragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
-
-        private val phrases = mutableListOf<Phrase>()
-        var numPages: Int = 0
-
-        fun setPhrases(phrases: List<Phrase>) {
-            with(this.phrases) {
-                clear()
-                addAll(phrases)
-            }
-
-            numPages = if (phrases.isEmpty()) {
-                1
-            } else {
-                ceil(phrases.size / maxPhrases.toDouble()).toInt()
-            }
-            notifyDataSetChanged()
+        VocableFragmentStateAdapter<Phrase>(fm, viewLifecycleOwner.lifecycle) {
 
+        override fun setItems(items: List<Phrase>) {
+            super.setItems(items)
             setPagingButtonsEnabled(phrasesAdapter.numPages > 1)
         }
 
         private fun setPagingButtonsEnabled(enable: Boolean) {
             binding?.let {
```

Comment:
You can alternatively do a binding?.apply here and then you wouldn't need to use "it" in the body

Reply:
Good point, we'll address these blocks going forward. There's multiple places we use this style.